### ISSUE TITLE
'Certificate path doesn't exist' when using DOCKER_HOST=unix:///var/run/docker.sock

### ISSUE THREAD

**Issue by francisdb** [docker-java/docker-java#480]
Switched to `3.0.0-RC1` after #419 was fixed
But now the config builder fails with setting
`export DOCKER_HOST=unix:///var/run/docker.sock`

`DockerClientConfig.createDefaultConfigBuilder().build()`

```
com.github.dockerjava.api.exception.DockerClientException: Certificate path (DOCKER_CERT_PATH) '/var/lib/jenkins/.docker/certs' doesn't exist. (DockerClientConfig.java:112)
[error] com.github.dockerjava.core.DockerClientConfig.checkDockerCertPath(DockerClientConfig.java:112)
[error] com.github.dockerjava.core.DockerClientConfig.<init>(DockerClientConfig.java:85)
[error] com.github.dockerjava.core.DockerClientConfig$DockerClientConfigBuilder.build(DockerClientConfig.java:431)
[error] com.whisk.docker.DockerKit$class.$init$(DockerKit.scala:10)
```

Am I correct that for sock communication there is no need for the certificate path check? At least on `2.x` this used to work.

**Comment by marcuslinke** [comment]
Probably you have set DOCKER_TLS_VERIFY=1. I'm not sure whether DOCKER_CERT_PATH should'nt be checked when using `DOCKER_HOST=unix:///var/run/docker.sock`. I think best option would be to improve error message here.

**Comment by marcuslinke** [comment]
@francisdb Something like 'Enabled TLS verification (DOCKER_TLS_VERIFY=1) but certifate path (DOCKER_CERT_PATH) '/var/lib/jenkins/.docker/certs' doesn't exist.'

**Comment by francisdb** [comment]
This using a default `Ubuntu 15.10` with `Docker version 1.10.0, build 590d5108` from `deb https://apt.dockerproject.org/repo ubuntu-wily main`

The only docker env variable that is set is `export DOCKER_HOST=unix:///var/run/docker.sock`

**Comment by marcuslinke** [comment]
Hmmm. OK. This is probably caused due to the default configuration of docker-java.properties (see https://github.com/docker-java/docker-java/blob/master/src/main/resources/docker-java.properties) that assumes secured tcp connection as preferred. 

Switching manually to `DOCKER_TLS_VERIFY=0` whether programmatically or via environment variable should solve it. Could you test that please?

**Comment by KostyaSha** [comment]
`'/var/lib/jenkins/.docker/certs' doesn't exist.'` is default directory. You should specify tls=0 if you want disable it.

**Comment by KostyaSha** [comment]
Also note that property file name was changed to docker-java.properties

**Comment by francisdb** [comment]
Ok the config issue is gone with that, but still somewhat unexpected that it by default checks certs when I ask a sock connection

**Comment by KostyaSha** [comment]
Do you have such variables in env? If yes, then it's expected to be used.

**Comment by francisdb** [comment]
everything worked correctly with only `export DOCKER_HOST=unix:///var/run/docker.sock` in 2.x
Now I have to add `DOCKER_TLS_VERIFY=0`, it's not a big issue but somewhat unexpected

**Comment by marcuslinke** [comment]
Maybe we have to rethink defaults in `src/main/resources/.docker-java.properties`...

**Comment by francisdb** [comment]
by the way, creating `/var/lib/jenkins/.docker/certs` brings me back to #419 https protocol is not supported (3.0.0-RC4)

**Comment by marcuslinke** [comment]
related: https://github.com/docker-java/docker-java/issues/576

**Comment by KostyaSha** [comment]
No need to bundle defaults in classpaths. If somebody doesn't like variables inheritance then it free to implement own clientconfig (or we didn't expose enough methods?).

**Comment by marcuslinke** [comment]
Closed via #590
